### PR TITLE
reference/system-databases: add description for `TIDB_ROW_ID_SHARDING_INFO`

### PR DESCRIPTION
**Pull request by bb7133**
### What is changed, added or deleted? 
Add description for the new `TIDB_ROW_ID_SHARDING_INFO` column in `information_schema`.`tables` table.

### What is the related PR or file link(s)? 

- [x] Reference link(s):https://github.com/pingcap/tidb/pull/13418 and https://github.com/pingcap/tidb/pull/14583

- [ ] This PR is to align with:
- [ ] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [x] dev (the latest development version)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [x] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: dev/reference/system-databases/information-schema.md [hunk 1]
```diff
@@ -477,6 +478,15 @@ SHOW TABLES
...
 ```
 
+表中的信息大部分定义自 MySQL，此外有两列是 TiDB 新增的：
+
+  * `TiDB_TABLE_ID`：标识表的内部 ID，该 ID 在一个 TiDB 集群内部唯一。
+  * `TIDB_ROW_ID_SHARDING_INFO`：标识表的 Sharding 类型，可能的值为：
+    - `"NOT_SHARDED"`：表未被 Shard。
+    - `"NOT_SHARDED(PK_IS_HANDLE)"`：一个定义了整型主键的表未被 Shard。
+    - `"PK_AUTO_RANDOM_BITS={bit_number}"`：一个定义了整型主键的表由于定义了 `AUTO_RANDOM` 而被 Shard。
+    - `"SHARD_BITS={bit_number}"`：表使用 `SHARD_ROW_ID_BITS={bit_number}` 进行了 Shard。
```
Comment (lilin90) [inline dev/reference/system-databases/information-schema.md]: Please remove extra spaces at the beginning of line 483~484, and use four-indentation for the 4 subitems. Then the lint failure will be fixed.

Comment (tangenta) [inline dev/reference/system-databases/information-schema.md]: How about `NULL`?
> nil: for tables that are not suitable(views, system tables).

Comment (bb7133) [reply dev/reference/system-databases/information-schema.md]: Done, PTAL

Comment (bb7133) [reply dev/reference/system-databases/information-schema.md]: Done, Thanks!

**Comment by bb7133** [comment]
PTAL @lilin90 @tangenta

File: dev/reference/system-databases/information-schema.md [hunk 2]
```diff
@@ -477,6 +478,16 @@ SHOW TABLES
   [LIKE 'wild']
 ```
 
+表中的信息大部分定义自 MySQL，此外有两列是 TiDB 新增的：
+
+* `TiDB_TABLE_ID`：标识表的内部 ID，该 ID 在一个 TiDB 集群内部唯一。
```
Comment (djshow832) [inline dev/reference/system-databases/information-schema.md]:
```suggestion
* `TIDB_TABLE_ID`：标识表的内部 ID，该 ID 在一个 TiDB 集群内部唯一。
```

Comment (bb7133) [reply dev/reference/system-databases/information-schema.md]: Addressed, thanks.

**Pull request merged**